### PR TITLE
UI:Add owner support for team

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
### Describe your changes :

I worked on adding support for the team owner.

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Frontend Preview (Screenshots) :
image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@shahsank3t, @darth-coder00 @ShaileshParmar11 @vivekratnavel

### CODE REVIEW

**Comment by harshach** [comment]
@Sachin-chaurasiya let's add tests for this functionality.
1. We should show the Add Users or delete user functionality to Admin or Owner of the team

**Comment by harshach** [comment]
@Sachin-chaurasiya for this release lets only allow user to be as owner. Team being owner of another team we should think through the use cases.
cc @sureshms

**Comment by Sachin-chaurasiya** [comment]
> @Sachin-chaurasiya let's add tests for this functionality.
> 
> 1. We should show the Add Users or delete user functionality to Admin or Owner of the team

@harshach, I have Added a unit test for the manage tab.
Also added check for the owner, so if a user is an owner or admin then they will be able to add a new user and delete the team.

**Review by darth-coder00 — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/src/pages/teams/index.test.tsx [hunk 1]
```diff
@@ -296,4 +302,19 @@ describe('Test Teams page', () => {
...
+
+    fireEvent.click(
+      assets,
+      new MouseEvent('click', {
+        bubbles: true,
+        cancelable: true,
+      })
+    );
+
+    expect(await findByText(container, /ManageTab/i)).toBeInTheDocument();
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/pages/teams/index.test.tsx:318]: Can we use test ids and check if the manage tab renders the save button or any other component that we expect to present in the Manage tab after it completes the initial fetch from API?

File: openmetadata-ui/src/main/resources/ui/src/pages/teams/index.tsx [hunk 2]
```diff
@@ -531,6 +565,33 @@ const TeamsPage = () => {
...
+          fetchCurrentTeam(res.data.name, true);
+        })
+        .catch((err: AxiosError) => {
+          reject();
+          const message = err.response?.data?.message;
+          showToast({
+            variant: 'error',
+            body:
+              message ??
+              `Error while updating owner for ${
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/pages/teams/index.tsx:587]: Can we move these error messages to a constants file?

File: openmetadata-ui/src/main/resources/ui/src/pages/teams/index.test.tsx (hunk 1 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/pages/teams/index.test.tsx:318]: @vivekratnavel , This all things are being covered in manageTab unit test. so here we are just mocking it.

**Pull request merged**